pgscm/templates/farmer/index.html

```diff
@@ -1,10 +1,82 @@
+            export_columns='0,1,2,3,4',
+            column_names=["farmer_code", "name", "gender", "type", "group_id"],
+            select2_class=g.c.SELECT_DEFAULT_ID) }}
+{%- endblock scripts %}
+{% block content_header -%}
+    <h1>
+        {{ _('PGS Farmer') }}
+        <small>{{ _('Managing farmers in PGS system') }}</small>
+    </h1>
+    <ol class="breadcrumb">
+        <li><a href="{{ lurl_for('main.index') }}"><i
+                class="fa fa-dashboard"></i> {{ _('Home') }}</a></li>
+        <li class="active"><a href="{{ lurl_for('farmer.index') }}"><i
+                class="fa fa-home"></i>{{ _('Farmer') }}</a></li>
+    </ol>
+{%- endblock content_header %}
+{% block content -%}
+    {{ render_btn('addBtn', 'btn-default', '', 'modal-add', _('Create New'), 'plus') }}
+    <div class="row">
+        <div class="col-xs-12">
+            <div class="box">
+                <div class="box-header">
+                    <h3 class="box-title">{{ _('PGS Farmers') }}</h3>
+                </div>
+                <!-- /.box-header -->
+                <div class="box-body">
+                    <table id="pgs_data"
+                           class="table table-bordered table-hover">
+                        <thead>
+                        <tr>
+                            <th>{{ _('Code') }}</th>
+                            <th>{{ _('Name') }}
+                            <th>{{ _('Gender') }}</th>
+                            <th>{{ _('Type') }}</th>
+                            <th>{{ _('Group') }}</th>
+                            <th>{{ _('Actions') }}</th>
+                        </tr>
+                        </thead>
+                        <tbody>
+                        {% for farmer in farmers %}
+                            <tr>
+                                <td>
+                                    <a href="#">
+                                        {{ farmer.farmer_code }}
+                                    </a>
+                                </td>
+                                <td><b>{{ farmer.name }}</b></td>
+                                <td>{{ genderType(farmer.gender).name }}</td>
+                                <td>{{ farmerType(farmer.type).name }}</td>
+                                <td>{{ farmer.group.name }}</td>
+                                <td class="actions_column">
+                                    {{ render_btn('editBtn', 'btn-xs', 'btn-info', 'modal-edit', '', 'edit') }}
+                                    {{ render_btn('delBtn', 'btn-xs', 'btn-danger', 'modal-delete', '', 'trash') }}
+                                </td>
+                            </tr>
+                        {% endfor %}
+                    </table>
+                </div>
+                <!-- /.box-body -->
+            </div>
+            <!-- /.box -->
+        </div>
+        <!-- /.col -->
+    </div>
+    <!-- /.row -->
+    {{ render_modal('add', 'modal', _('Create New')) }}
```

Comment:
the button ID, modal ID now have a constant ID calling via g.c.BTNADD_ID, check my newest example code.